include/zephyr/senss/senss.h

```diff
@@ -0,0 +1,384 @@
+		struct {
+			uint8_t major;
+			uint8_t minor;
+			uint8_t hotfix;
+			uint8_t build;
+		};
+	};
+};
+
+#define SENSS_SENSOR_INVALID_HANDLE (-1)
+#define SENSS_INDEX_ALL (-1)
+
+#define SENSS_SENSOR_VERSION(_major, _minor, _hotfix, _build)           \
+	((_major) << 24 | (_minor) << 16 | (_hotfix) << 8 | (_build))
+
+
+/**
+ * @brief Sensor flag indicating if this sensor is on event reporting data.
+ *
+ * Reporting sensor data when the sensor event occurs, such as a motion detect sensor reporting
+ * a motion or motionless detected event.
+ */
+#define SENSS_SENSOR_FLAG_REPORT_ON_EVENT			BIT(0)
+
+/**
+ * @brief Sensor flag indicating if this sensor is on change reporting data.
+ *
+ * Reporting sensor data when the sensor data changes.
+ *
+ * Exclusive with  \ref SENSS_SENSOR_FLAG_REPORT_ON_EVENT
+ */
+#define SENSS_SENSOR_FLAG_REPORT_ON_CHANGE			BIT(1)
+
+
+/**
+ * @brief Sensor state.
+ *
+ */
+enum senss_sensor_state {
+	SENSS_SENOSR_STATE_NOT_READY = 1,
+	SENSS_SENOSR_STATE_READY = 2,
+};
+
+
+/**
+ * @struct senss_sensor_info
+ * @brief Sensor basic constant information
+ *
+ */
+struct senss_sensor_info {
+	/** Name of the sensor instance */
+	const char *name;
+
+	/** Friendly name of the sensor instance */
+	const char *friendly_name;
+
+	/** Vendor name of the sensor instance */
+	const char *vendor;
+
+	/** Model name of the sensor instance */
+	const char *model;
+
+	/** Sensor type */
+	int32_t type;
+
+	/** Sensor index for multiple sensors under same sensor type */
+	int32_t sensor_index;
+
+	/** Sensor flags */
+	uint32_t flags;
+
+	/** Minimal report interval in micro seconds  */
+	uint32_t minimal_interval;
+
+	/** Sensor version */
+	struct senss_sensor_version version;
+};
+
+
+
+/**
+ * @brief Initialize sensor subsystem runtime.
+ *
+ * This will allocate internal resources, such as create internal schedule
+ * task, create internal management data structures.
+ *
+ * It will also enumerate all registered sensor instances in sensor subsystem,
+ * build sensors report relational data model, if a virtual sensor's required dependency
+ * sensor type not found, will return error.
+ * Then call each sensor instances' init callback API.
+ *
+ * @return 0 on success or negative error value on failure.
+ *
+ */
+int senss_init(void);
+
+ /**
+  * @brief De-initialize sensor subsystem runtime.
+  *
+  * This will release all dynamically allocated internal resources, and call
+  * all sensor instances' deinit callback API.
+  *
+  * @return 0 on success or negative error value on failure.
+  */
+int senss_deinit(void);
+
+ /**
+  * @brief Get all supported sensor instances' information.
+  *
+  * This API just returned read only information of sensor instances,
+  * no side effect to sensor instances.
+  *
+  * @param **info For receiving sensor instances' information array pointer.
+  *
+  * @return Number of sensor instances, 0 returned if no sensors.
+  */
+int senss_get_sensors(const struct senss_sensor_info **info);
```

Comment:
nit: return `int` for error and pass a pointer `size_t *num_sensors`. If for some reason the subsystem's `deinit` function was called, then `get_sensors` will fail and returning `0` isn't informative

Reply:
modified senss_get_sensors() according to review comments, thanks.